test/Isolate.t.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+/// @dev A test contract to ensure developers are using `--isolate` flag when running forge test
+contract IsolateTest is Test {
+    StorageLib storageLib;
+
+    function setUp() public {
+        storageLib = new StorageLib();
+    }
+
+    function testIsolateTest() public {
+        // tstore key: 1 with value :2
+        storageLib.tstore(1, 2);
+
+        // toload key: 1
+        uint256 val = storageLib.tload(1);
+
+        // If the test is run with `--isolate` flag, the value should be 0
+        // as --isolate run each top level call as seperate transaction, so tload will return 0
+        assertEq(val, 0, "did you forget to use --isolate flag for 'forge test'?");
```

Comment:
This seems a bit counter-intuitive for me. Just to confirm, does that mean:

1. Both `storageLib.tstore(1, 2);` and `uint256 val = storageLib.tload(1);` are separate txs
2. If there are more external calls inside for example `storageLib.tload` it will behave like normal function call instead of stacking up another new tx

Reply:
> 1. Both `storageLib.tstore(1, 2);` and `uint256 val = storageLib.tload(1);` are separate txs

Yes, they will be seperate transaction

> 2. If there are more external calls inside for example `storageLib.tload` it will behave like normal function call instead of stacking up another new tx

Yes. the seperate transaction is only from top level call. 


Do you have a recommendation for the comment on the test or assert error message so it can be clearer? we can update to it

Reply:
ehhh, maybe we can mention it in README or FAQ somewhere. That should avoid most unnecessary debugging if someone encounters it 😂

Reply:
updated readme with details 😁